src/osdc/Objecter.cc

```diff
@@ -451,12 +451,12 @@ void Objecter::shutdown()
     cop->put();
   }
 
-  if (tick_event) {
-    if (timer.cancel_event(tick_event)) {
+  if (tick_event.read()) {
+    if (timer.cancel_event(tick_event.read())) {
```

Comment:
Thank you for catching that. I think it's better just to drop that assert in tick() entirely. It's fragile and doesn't really guard against anything except someone calling 'tick' directly.